candle-transformers/src/models/xlm_roberta.rs

```diff
@@ -0,0 +1,545 @@
+    pub fn new(cfg: &Config, vb: VarBuilder) -> Result<Self> {
+        let roberta = XLMRobertaModel::new(cfg, vb.pp("roberta"))?;
+        let lm_head = XLMRobertaLMHead::new(cfg, vb.pp("lm_head"))?;
+        Ok(Self { roberta, lm_head })
+    }
+
+    pub fn forward(
+        &self,
+        input_ids: &Tensor,
+        attention_mask: &Tensor,
+        token_type_ids: &Tensor,
+        past_key_value: Option<(&Tensor, &Tensor)>,
+        encoder_hidden_states: Option<&Tensor>,
+        encoder_attention_mask: Option<&Tensor>,
+    ) -> Result<Tensor> {
+        let hidden_states = self.roberta.forward(
+            input_ids,
+            attention_mask,
+            token_type_ids,
+            past_key_value,
+            encoder_hidden_states,
+            encoder_attention_mask,
+        )?;
+        let lm_logits = self.lm_head.forward(
+            &hidden_states,
+            &self
+                .roberta
+                .embeddings
+                .word_embeddings
+                .embeddings()
+                .t()?
+                .unsqueeze(0)?,
+        )?;
+        Ok(lm_logits)
+    }
+}
+
+struct XLMRobertaClassificationHead {
+    dense: Linear,
+    out_proj: Linear,
+}
+
+impl XLMRobertaClassificationHead {
+    fn new(num_labels: usize, cfg: &Config, vb: VarBuilder) -> Result<Self> {
+        let dense = linear(cfg.hidden_size, cfg.hidden_size, vb.pp("dense"))?;
+        let out_proj = linear(cfg.hidden_size, num_labels, vb.pp("out_proj"))?;
+        Ok(Self { dense, out_proj })
+    }
+
+    fn forward(&self, hidden_states: &Tensor) -> Result<Tensor> {
+        let cls_states = hidden_states.get_on_dim(1, 0)?.contiguous()?;
+        let hidden_states = self.dense.forward(&cls_states)?;
+        let hidden_states = candle_nn::Activation::GeluPytorchTanh.forward(&hidden_states)?;
+        let hidden_states = self.out_proj.forward(&hidden_states)?;
+        Ok(hidden_states)
+    }
+}
+
+pub struct XLMRobertaForSequenceClassification {
+    roberta: XLMRobertaModel,
+    classifier: XLMRobertaClassificationHead,
+}
+
+impl XLMRobertaForSequenceClassification {
+    pub fn new(num_labels: usize, cfg: &Config, vb: VarBuilder) -> Result<Self> {
+        let roberta = XLMRobertaModel::new(cfg, vb.pp("roberta"))?;
+        let classifier = XLMRobertaClassificationHead::new(num_labels, cfg, vb.pp("classifier"))?;
+        Ok(Self {
+            roberta,
+            classifier,
+        })
+    }
+
+    pub fn forward(
+        &self,
+        input_ids: &Tensor,
+        attention_mask: &Tensor,
+        token_type_ids: &Tensor,
+    ) -> Result<Tensor> {
+        let hidden_states =
+            self.roberta
+                .forward(input_ids, attention_mask, token_type_ids, None, None, None)?;
+        self.classifier.forward(&hidden_states)
+    }
+}
+
+fn prepare_4d_attention_mask(
+    mask: &Tensor,
+    dtype: DType,
+    tgt_len: Option<usize>,
+) -> Result<Tensor> {
+    let bsz = mask.dims()[0];
```

Comment:
Use `let bsz = mask.dim(0)?` instead, this will make a proper error rather than a panic if the shape is not correct. Same for the line below.